avocado/core/task/statemachine.py

```diff
@@ -280,6 +280,14 @@ async def monitor(self):
                 runtime_task.status = 'FINISHED'
             except asyncio.TimeoutError:
                 runtime_task.status = 'FINISHED W/ TIMEOUT'
+                await asyncio.wait_for(self._spawner.terminate_task(runtime_task), None)
```

Comment:
I don't quite follow why you call `wait_for()` without a timeout.  Couldn't this simply be `await self._spawner.terminate_task(runtime_task)`?

BTW, eventually we'll also need to address timeouts *while* terminating tasks, so using `wait_for()` (or something like it) would be valid then.